### PR TITLE
Add architecture diagrams

### PR DESCRIPTION
**Pull request by luwes**
Docs link: 
- https://github.com/luwes/media-chrome/blob/ui-controller-media-diagram/docs/architecture-diagrams.md

Figma links:
- https://www.figma.com/file/JfpS4VVSJgvywPHEYAr7Ie/Media-Chrome-Diagrams?node-id=0%3A1
- https://www.figma.com/file/JfpS4VVSJgvywPHEYAr7Ie/Media-Chrome-Diagrams?node-id=26%3A84

### CODE REVIEW

**Comment by heff** [comment]
These are cool! Is the goal to publish these in the repo?

> In case 2. below the UI element would not be nested under the <media-controller>
What do we mean by "below the UI element" here?

image

You can't use the media-controller attr on a generic div today (unless I missed something). You would need to use the `associateElement` API to register the div with the controller. If the div is a child of the controller or an associated element, it should only need the media-chrome-attribute attr.

**Comment by luwes** [comment]
Added the markdown version 
https://github.com/muxinc/media-chrome/blob/26a41ce799d1ca2ded7e00bd5d71cb9de1587cf7/docs/architecture-diagrams.md

File: docs/architecture-diagrams.md
Comment (cjpillsbury) [inline docs/architecture-diagrams.md:43]: Fine for now/forever (unless @heff has concerns), but the "Send" arrow from the elements is slightly lying, since it's bubbling to the `media-control-bar` ("associated element") and that's what is getting picked up by the event handlers in the `MediaController` instance. Since you discuss this in the description ☝️, I'm ok with this visual diagram.

File: docs/architecture-diagrams.md
Comment (cjpillsbury) [inline docs/architecture-diagrams.md:8]: consider adding 
> The `MediaController` may receive these events in one of two ways:
(leave it up to you for final discretion to add)

File: docs/architecture-diagrams.md
Comment (cjpillsbury) [inline docs/architecture-diagrams.md:22]: I think we should introduce the term "media state receiver" here. Something like:
> Any **associated element** or any of its descendants can receive media state from the `MediaController`, as long as the elements are identifiable as something that _*should*_ receive media state (aka identifiable as a **media state receiver**). Elements are identified as media state receivers in one of two ways:

Then you can discuss the `observedAttributes` and `[media-chrome-attributes]`

File: docs/architecture-diagrams.html
Comment (cjpillsbury) [inline docs/architecture-diagrams.html:1]: Fine to keep for now to make iteration easier, but let's plan on deleting this before merging to `#main`

File: docs/architecture-diagrams.md
Comment (cjpillsbury) [inline docs/architecture-diagrams.md:1]: Let's add a link to this doc somewhere in the main `README.md`

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
Some small feedback.

**Comment by luwes** [comment]
new link: https://github.com/muxinc/media-chrome/blob/2a0ef1f99a0cc7ac184f94b13d6ed7f7b845fbb2/docs/architecture-diagrams.md

**Review by cjpillsbury — APPROVED** [review]

File: docs/architecture-diagrams.md
Comment (heff) [reply docs/architecture-diagrams.md:66]: Yeah, it's probably fine for this high level understanding of how the system works. Otherwise it feels like we'd have to break out of the simple send/receive concepts and get into the specifics of emitting events, listeners, calling functions, etc.

**Review by heff — APPROVED** [review]
These are great. 👍  Thanks @luwes.

**Pull request merged**
